src/core/Akka/Actor/Scheduler/HashedWheelTimerScheduler.cs

```diff
@@ -110,21 +115,19 @@ private static int NormalizeTicksPerWheel(int ticksPerWheel)
 
         private readonly HashSet<SchedulerRegistration> _unprocessedRegistrations = new HashSet<SchedulerRegistration>();
         private readonly HashSet<SchedulerRegistration> _rescheduleRegistrations = new HashSet<SchedulerRegistration>();
-
-        private Thread _worker;
+        private readonly Stopwatch _sleepWatch = new Stopwatch();
 
         private void Start()
         {
             if (_workerState == WORKER_STATE_STARTED) { } // do nothing
             else if (_workerState == WORKER_STATE_INIT)
             {
-                _worker = new Thread(Run) { IsBackground = true };
 #pragma warning disable 420
                 if (Interlocked.CompareExchange(ref _workerState, WORKER_STATE_STARTED, WORKER_STATE_INIT) ==
 #pragma warning restore 420
                     WORKER_STATE_INIT)
                 {
-                    _worker.Start();
+                    Task.Run(Run);
```

Comment:
I think this might be why we're having issues on the MNTR - the design before called for the `Scheduler` to run on a dedicated thread and periodically sleep / wait itself. This was intentional - we wanted the dedicated thread so other items piling up inside the `TaskScheduler`'s queue couldn't affect time-sensitive jobs performed by the scheduler.

So here's my suggestion: - use a custom `TaskScheduler` that runs on a single, dedicated thread (I think you can find an example inside the `ForkJoinExecutor` code inside the Akka.Dispatch namespace) and have the scheduler run on that. No more work queue issues - the only competition will be for the CPU itself as the operating system scheduler level. Moreover, we can also use `await` and `Task.Delay` instead of sleep to see if that reduces CPU utilization coming from the scheduler (wasted cycles on `Thread.Sleep`) - and if that doesn't work, we can go back to using `Thread.SpinWait`.

Does that sound reasonable?

Reply:
The first part is clear, and that's really good idea, I will work on this tomorrow.

But did not get why should be get back to `Task.Delay`? Most of the time scheduler needs to wait less then `10ms` for the next tick - and `Task.Delay` will always give us more sleep time (the same +10% as it was). 
What can be done is still use `Task.Delay` for pauses more then `10ms` - they are possible if scheduler tick duration will be set to something >10ms via `akka.scheduler.tick-duration` setting.
@Aaronontheweb Or we should get back to `Task.Delay` because of some other benefits?

Reply:
> Or we should get back to Task.Delay because of some other benefits?

Ah, nah, nevermind - for some reason I assumed that `Task.Delay` could use an I/O completion port and not use CPU when idling... You'd need to look at the reference source to be sure. Thread.SpintWait is probably best.